src/Container.ts

```diff
@@ -418,10 +417,19 @@ export class Container<Services = {}> {
    *            specifying these dependencies.
    * @returns A new Container instance containing the newly created service, allowing for method chaining.
    */
-  providesClass = <Token extends TokenType, Service, Tokens extends readonly ValidTokens<Services>[]>(
+  providesClass = <
+    Token extends TokenType,
+    Tokens extends readonly ValidTokens<Services>[],
+    Class extends {
```

Comment:
Nice, does it mean we can remove InjectableClass and ClassInjectable now?

Reply:
yes, removed that.